pandas/core/generic.py

```diff
@@ -4173,6 +4173,14 @@ def _drop_axis(
 
         # Case for non-unique axis
         else:
+            if (
```

Comment:
should this case only be an issue starting on L4193? e.g. level is None there

Reply:
Yep, this is correct. Question is what to do when level is None.

I think ``~axis.get_level_values(level).isin(labels)`` is the way to go, so I check before reaching the if else block if level is None to set it to 0 in this case

Reply:
i think setting to level = 0 is fine, i would just move the check *inside* the block on L4193 as its not going to be true where it is if level is not None in the first place.

Reply:
This would work as an alternative